packages/react-dom/src/__tests__/ReactDOMEventPropagation-test.js

```diff
@@ -0,0 +1,1327 @@
+    document.body.appendChild(container);
+    OuterReactDOM.render(tree, container);
+  }
+
+  describe('bubbling events', () => {
+    // This test will fail in legacy mode (only used in WWW)
+    // because we emulate the React 16 behavior where
+    // the click handler is attached to the document.
+    // @gate !enableLegacyFBSupport
+    it('onClick', () => {
+      testNativeBubblingEvent({
+        type: 'div',
+        reactEvent: 'onClick',
+        nativeEvent: 'click',
+        dispatch(node) {
+          node.click();
+        },
+      });
+    });
+
+    it('onKeyPress', () => {
+      testNativeBubblingEvent({
+        type: 'input',
+        reactEvent: 'onKeyPress',
+        nativeEvent: 'keypress',
+        dispatch(node) {
+          node.dispatchEvent(
+            new KeyboardEvent('keypress', {
+              keyCode: 13,
+              bubbles: true,
+              cancelable: true,
+            }),
+          );
+        },
+      });
+    });
+
+    it('onMouseDown', () => {
+      testNativeBubblingEvent({
+        type: 'button',
+        reactEvent: 'onMouseDown',
+        nativeEvent: 'mousedown',
+        dispatch(node) {
+          node.dispatchEvent(
+            new MouseEvent('mousedown', {
+              bubbles: true,
+              cancelable: true,
+            }),
+          );
+        },
+      });
+    });
+
+    it('onTouchStart', () => {
+      testNativeBubblingEvent({
+        type: 'div',
+        reactEvent: 'onTouchStart',
+        nativeEvent: 'touchstart',
+        dispatch(node) {
+          node.dispatchEvent(
+            new Event('touchstart', {
+              bubbles: true,
+              cancelable: true,
+            }),
+          );
+        },
+      });
+    });
+
+    it('onWheel', () => {
+      testNativeBubblingEvent({
+        type: 'div',
+        reactEvent: 'onWheel',
+        nativeEvent: 'wheel',
+        dispatch(node) {
+          node.dispatchEvent(
+            new Event('wheel', {
+              bubbles: true,
+              cancelable: true,
+            }),
+          );
+        },
+      });
+    });
+
+    it('onSubmit', () => {
+      testNativeBubblingEvent({
+        type: 'form',
+        reactEvent: 'onSubmit',
+        nativeEvent: 'submit',
+        dispatch(node) {
+          const e = new Event('submit', {
+            bubbles: true,
+            cancelable: true,
+          });
+          node.dispatchEvent(e);
+        },
+      });
+    });
+
+    it('onReset', () => {
+      testNativeBubblingEvent({
+        type: 'form',
+        reactEvent: 'onReset',
+        nativeEvent: 'reset',
+        dispatch(node) {
+          const e = new Event('reset', {
+            bubbles: true,
+            cancelable: true,
+          });
+          node.dispatchEvent(e);
+        },
+      });
+    });
+
+    it('onFocus', () => {
+      testNativeBubblingEvent({
+        type: 'input',
+        reactEvent: 'onFocus',
+        nativeEvent: 'focusin',
```

Comment:
This one is interesting. It's in the bubbling category because it's effectively an alias for the browser `onfocusin` which *does* bubble. So it's React event name that's a misnomer.